### DISCUSSION TITLE
Updating fragment inside a form

### DISCUSSION THREAD

**Discussion by dappa** [unpoly/unpoly#348] [Q&A] [answered]
I'm having some issues conceptualizing the idea of pulling in HTML in the middle of a form and how to do it in Unpoly from a specific field like in the example below:
![Screenshot 2022-01-28 133208](https://user-images.githubusercontent.com/2115659/151547590-f6dc03c7-b04b-488d-ac65-527d2f4eab50.png)

Here I want to fetch HTML when clicking the green button, and insert it below. Normally I'd use `up-submit` in a form, but seeing as we're already in a form, I can't do that or I'd have to submit the entire form and have some additional logic around this one area when validating.

In HTMX I'd just add a `hx-get` attribute to the green button, a `hx-include` to the input, and assign a target, but I'm not sure of Unpoly has anything similar?

**Comment by adam12** [comment, marked as answer]
There's no directive that I'm aware of that supports what you want, but I might be inclined to just add some JavaScript to the `<button>` here (you could of course make your own `up.compiler` and setup a different event listener.

```html
<button type="button" onclick="up.validate('input[name=registration_number]', {target: '#registration_info'})">
  Search
</button>

<div id="registration_info"></div>
```

On the server, check the `X-Up-Validate` header (or the appropriate method in the Unpoly library you're using), and then re-render the form with the data inside the target.
